src/runtime/simulator.cu

```diff
@@ -114,8 +80,18 @@ void Simulator::strategy_search_task(const Task *task,
   // Realm::Cuda::GPUFBMemory* memFBImpl = (Realm::Cuda::GPUFBMemory*) memImpl;
   // off_t offset = memFBImpl->alloc_bytes_local(model->config.simulator_work_space_size);
   // void* base_ptr = memFBImpl->get_direct_ptr(offset, 0);
+  MachineModel *machine;
+  if (model->config.machine_model_version == 0) {
+    machine = (MachineModel *) new MachineModel_old(model->config.numNodes, model->config.workersPerNode, gpu_mem.capacity());
+  }
+  else if (model->config.machine_model_version == 1 and !model->config.machine_model_file.empty()) {
+    machine = (MachineModel *) new MachineModel_new(model->config.machine_model_file, gpu_mem.capacity());
+  }
+  else {
+    assert(false && "machine model creation error");
```

Comment:
More useful error message here since this is not unlikely to happen?

Reply:
Fixed.